Python/HAT/hypergraph.py

```
@@ -365,75 +473,58 @@ def set_adjacency_tensor(self, AT=None):
             # Populate adjacency tensor based on edge list
             for i, edge in enumerate(self.edge_list):
-                if len(edge) == k:
-                    # print(f"{edge=}")
-                    # print(f"{tuple(edge)=}")
+                if len(edge) == self.order:
                     # Use multi-dimensional indexing to set entries for the k-node hyperedge
-                    self._adjacency_tensor[tuple(edge)] = 1
+                    for nodes_indices in permutations(edge):
+                        self._adjacency_tensor[tuple(nodes_indices)] = 1
                 else:
                     warnings.warn(f"Edge {i} does not have exactly {k} nodes; skipping.")
```

Ruff (0.8.0)

No explicit `stacklevel` keyword argument found

(B028)